### DISCUSSION TITLE
Large File Upload into parts/chunks on S3 via React native

### DISCUSSION THREAD

**Discussion by mobileexpert1** [aws/aws-sdk-js-v3#3730] [Q&A]
Which is the best way to upload a large file (Video and Audio) into parts on S3 bucket ?

Do we have any specific SDK available  for React native to get file upload only?

I want to upload file up to 3 GB (max.)

**Comment by aBurmeseDev** [comment]
Hi @mobileexpert1 - thanks for reaching out and apologies for delay.

Yes, you could use Javascript SDK with React native to upload file to your S3 bucket. If you're dealing with large files, I'd use [CreateMultipartUploadCommand](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/command/CreateMultipartUploadCommand/) operation from SDK. Here are a few blog posts about this topic for your reference:
- https://aws.amazon.com/blogs/compute/uploading-large-objects-to-amazon-s3-using-multipart-upload-and-transfer-acceleration/
- https://repost.aws/knowledge-center/s3-upload-large-files

Let me know if you need further assist with this, I'd be happy to help!
John
